### PR TITLE
optional getLogDir.

### LINKED ISSUE #44579: [RFC] Make Log dir nullable
### Description

In context of containers - one of best practices is to write logs in stdout and stderr, rather than in files. 
unfortunately - log dir is required - even if not used.

Proposal is to either add new interface like `LoggableKernel` which requires `getLogDir` or make `getLogDir` nullable. In both cases: disable checks on logs dir.

### Example

_No response_

### PR DESCRIPTION
**Pull request by Warxcell**
| Q             | A
| ------------- | ---
| Branch?       | 6.1 for features
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | yes
| Tickets       | Fix #44579 
| License       | MIT
| Doc PR        | symfony/symfony-docs#...  TODO

### CODE REVIEW

File: src/Symfony/Component/HttpKernel/Kernel.php [hunk 1]
```diff
@@ -622,6 +626,7 @@ protected function getKernelParameters(): array
      */
     protected function buildContainer(): ContainerBuilder
     {
+        // TODO: when removing getLogDir from KernelInterface, add check for LoggableKernelInterface
         foreach (['cache' => $this->getCacheDir(), 'build' => $this->warmupDir ?: $this->getBuildDir(), 'logs' => $this->getLogDir()] as $name => $dir) {
```
Comment (ro0NL) [inline src/Symfony/Component/HttpKernel/Kernel.php:630]: i'd be curious if we could just remove initializing the log directory here.

monolog creates it lazy AFAIK for stream handlers.

**Comment by IonBazan** [comment]
Please remove the `.idea` directory and squash your commits afterwards. You can add it to your global `.gitignore` file: https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files#configuring-ignored-files-for-all-repositories-on-your-computer

**Comment by Warxcell** [comment]
> Please remove the `.idea` directory and squash your commits afterwards. You can add it to your global `.gitignore` file: https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files#configuring-ignored-files-for-all-repositories-on-your-computer

Can we ignore them on project level, because I have some projects that actually commits .idea

**Comment by IonBazan** [comment]
You can always put it in your `.git/info/exclude` file as explained on the link above.

**Comment by nicolas-grekas** [comment]
I was surprised to read the attached patch. I was expecting that getLogDir could be allowed to return null. Not that we could remove it.

Right now, I still think that making getLogDir nullable is a better path forward.

**Comment by nicolas-grekas** [comment]
I propose #45640 instead.

**Comment by nicolas-grekas** [comment]
Just a wild idea: what about adding an attribute on the getLogDir method: `#[IgnoreOnInvalid]`?

**Comment by nicolas-grekas** [comment]
@Warxcell WDYT? In case you missed it, I closed my proposal.

**Comment by Warxcell** [comment]
> @Warxcell WDYT? In case you missed it, I closed my proposal.

what about returning NULL? Attribute looks even worse to me.

**Comment by nicolas-grekas** [comment]
This has been proposed and discussed a bit in eg https://github.com/symfony/symfony/pull/45640#issuecomment-1080021817, and it doesn't look bright...

**Comment by nicolas-grekas** [comment]
But honeslty, I'm fine closing as "won't fix" also. For ppl that don't want any extra dir, they could make getLogDir return getCacheDir. :see_no_evil:

**Comment by Warxcell** [comment]
Well with LoggableKernelInterface we have forward-compatibility path. Close if you wish so. In our case, we point it to `/tmp` as work-around.

**Comment by nicolas-grekas** [comment]
LoggableKernelInterface feels like pushing a problem that a minority has to the majority. Maybe I'm wrong and this isn't an issue?

**Comment by Warxcell** [comment]
Ok, what if we create `Symfony\Component\HttpKernel\LoggableKernel extends Symfony\Component\HttpKernel\Kernel implements LoggableKernelInterface` 
that contains implementation of interface.

```
public function getLogDir()
{
    return $this->getProjectDir().'/var/log';
}
```

And then create-project command will create local Kernel that extends by default `LoggableKernel` and anyone can opt-out by extending `Symfony\Component\HttpKernel\Kernel`.  Others will have to change single line of code.

This is my last resort on this. :) If we don't have agreement - just close it.

**Comment by fabpot** [comment]
Closing as it's really just a small inconvenience and we don't seem to find a consensus here.

**Pull request closed without merging**